### PR TITLE
add senta

### PR DESCRIPTION
**Pull request by Steffy-zxf**
add senta (cnn, gru, bilstm, lstm and bow) as module v2

### CODE REVIEW

File: modules/text/sentiment_analysis/senta_bilstm/resources/module.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+from senta_bilstm_python.net import bilstm_net
+from senta_bilstm_python.processor import load_vocab, preprocess, postprocess
+
+
+class DataFormatError(Exception):
+    def __init__(self, *args):
+        self.args = args
+
+
+class Senta_BiLSTM(hub.Module):
```
Comment (ZeyuChen) [inline modules/text/sentiment_analysis/senta_bilstm/resources/module.py]: How about use SentaBiLSTM, and SentaLSTM, SentaBoW, SentaCNN as standard naming style

Comment (ZeyuChen) [reply modules/text/sentiment_analysis/senta_bilstm/resources/module.py]: And is it possible to unifiy 4 Senta module into single one Senta, then use a new param="bilstm, cnn bow" to control its structure?

File: modules/text/sentiment_analysis/senta_bilstm/module.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        elif texts == [] and isinstance(data, dict) and isinstance(
+                data.get('text', None), list) and data['text']:
+            predicted_data = data["text"]
+        else:
+            raise ValueError(
+                "The input data is inconsistent with expectations.")
+
+        predicted_data = self.to_unicode(predicted_data)
+        if not self.lac:
+            self.lac = hub.Module(name="lac")
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/senta_bilstm/module.py]: 修改为属性获取的方式，同porn det

File: modules/text/sentiment_analysis/senta_bilstm/module.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+        processed_results = preprocess(self.lac, predicted_data, self.word_dict,
+                                       use_gpu)
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/senta_bilstm/module.py]: 应该支持batch_size

File: modules/text/sentiment_analysis/senta_bilstm/module.py [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+        except:
+            use_gpu = False
+        if use_gpu:
+            gpu_config = AnalysisConfig(
+                os.path.join(self.directory, "infer_model"))
+            gpu_config.disable_glog_info()
+            gpu_config.enable_use_gpu(memory_pool_init_size_mb=500, device_id=0)
+            self.gpu_predictor = create_paddle_predictor(gpu_config)
+
+    def context(
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/senta_bilstm/module.py]: 接口直接写为一行就行

File: modules/text/sentiment_analysis/senta_bilstm/resources/module.py (hunk 1 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_bilstm/resources/module.py]: 1. As for class name, I've already updated.
2. As for code reuse, now the module v2 haven't adapted the situation like senta well. I will inherit the ModuleV2 class  by NLPModuleV2 Class next time to fix this.

File: modules/text/sentiment_analysis/senta_bilstm/net.py [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+        param_attr=fluid.ParamAttr(name="@HUB_senta_bilstm@fc_2.w_0"),
+        bias_attr=fluid.ParamAttr(name="@HUB_senta_bilstm@fc_2.b_0"))
+    # softmax layer
+    prediction = fluid.layers.fc(
+        input=fc1,
+        size=class_dim,
+        act='softmax',
+        param_attr=fluid.ParamAttr(name="@HUB_senta_bilstm@fc_3.w_0"),
+        bias_attr=fluid.ParamAttr(name="@HUB_senta_bilstm@fc_3.b_0"))
+    cost = fluid.layers.cross_entropy(input=prediction, label=label)
```
[the remark this answers is not in the document]
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_bilstm/net.py]: 已去除

File: modules/text/sentiment_analysis/senta_bilstm/net.py [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+    avg_cost = fluid.layers.mean(x=cost)
+    acc = fluid.layers.accuracy(input=prediction, label=label)
+    return avg_cost, acc, prediction, fc1
+
+
+def bow_net(data,
```
[the remark this answers is not in the document]
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_bilstm/net.py]: 已去除

File: modules/text/sentiment_analysis/senta_bilstm/processor.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            if word in word_dict:
+                _index = word_dict[word]
+            else:
+                _index = unk_id
+            result_i['processed'].append(_index)
+        result.append(result_i)
+    return result
+
+
+def postprocess(
```
[the remark this answers is not in the document]
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_bilstm/processor.py]: 已修改

File: modules/text/sentiment_analysis/senta_bilstm/module.py [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+    def __init__(self, *args):
+        self.args = args
+
+
+@moduleinfo(
+    name="senta_bilstm",
+    version="1.1.0",
+    summary="Baidu's open-source Sentiment Classification System.",
+    author="baidu-nlp",
+    author_email="paddle-dev@baidu.com",
```
[the remark this answers is not in the document]
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_bilstm/module.py]: 已修改

File: modules/text/sentiment_analysis/senta_bilstm/module.py (hunk 2 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_bilstm/module.py]: 已修改

File: modules/text/sentiment_analysis/senta_bilstm/module.py (hunk 3 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_bilstm/module.py]: 已修改

File: modules/text/sentiment_analysis/senta_bilstm/module.py (hunk 4 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_bilstm/module.py]: 已修改

File: modules/text/sentiment_analysis/senta_bow/module.py [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+    def __init__(self, *args):
+        self.args = args
+
+
+@moduleinfo(
+    name="senta_bow",
+    version="1.1.0",
+    summary="Baidu's open-source Sentiment Classification System.",
+    author="baidu-nlp",
+    author_email="paddle-dev@baidu.com",
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/senta_bow/module.py]: email置空

File: modules/text/sentiment_analysis/senta_lstm/module.py [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+        except:
+            use_gpu = False
+        if use_gpu:
+            gpu_config = AnalysisConfig(
+                os.path.join(self.directory, "infer_model"))
+            gpu_config.disable_glog_info()
+            gpu_config.enable_use_gpu(memory_pool_init_size_mb=500, device_id=0)
+            self.gpu_predictor = create_paddle_predictor(gpu_config)
+
+    def context(
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/senta_lstm/module.py]: 写为一行

File: modules/text/sentiment_analysis/senta_lstm/module.py [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+    @property
+    def word_seg_module(self):
+        """
+        lac module
+        """
+        if not self._word_seg_module:
+            self._word_seg_module = hub.Module(name="lac")
+        return self._word_seg_module
+
+    def _set_config(self, ):
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/senta_lstm/module.py]: 多余逗号

File: modules/text/sentiment_analysis/senta_bow/module.py (hunk 9 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_bow/module.py]: done

File: modules/text/sentiment_analysis/senta_lstm/module.py (hunk 11 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_lstm/module.py]: done

File: modules/text/sentiment_analysis/senta_lstm/module.py (hunk 10 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/senta_lstm/module.py]: done

**Pull request merged**